### PR TITLE
HPCC-20089 Rationalize and consolidate the developer documentation

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman @JamesDeFabia please review.

Note, this is complete - many of these document could do with revising - e.g. I have added a list of some of the items to the style guide.  However I think that should be done as a separate PR.  This at least puts most of the documentation in once place and makes it easier to find.

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Minor comments inline, but overall, this is a great reorganization.

File: devdoc/Development.rst [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+https://github.com/hpcc-systems/HPCC-Platform/blob/master/CONTRIBUTORS for information about how to set up a GitHub
+fork of the project through which pull-requests can be made.
+
+********************************
+Building the system from sources
+********************************
+
+Requirements
+============
+The HPCC platform requires a number of third party tools and libraries in order to build.  The `HPCC Wiki`_ contains the
```
[the remark this answers is not in the document]
Comment (ghalliday) [reply devdoc/Development.rst]: It is a link - see the last line of the file.

File: devdoc/Development.rst [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    ./Debug/bin/eclagent -selftest
+
+You can also run the Dali regression self-tests::
+
+    ./Debug/bin/daregress localhost
+
+Regression Tests
+================
+
+**MORE** Completely out of date - needs rewriting.
```
[the remark this answers is not in the document]
Comment (ghalliday) [reply devdoc/Development.rst]: yes, something to look at later.

**Comment by ghalliday** [comment]
@JamesDeFabia you need to mention someone when you have finished.
@richardkchapman please review/merge

File: devdoc/Development.rst [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    #    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    #    See the License for the specific language governing permissions and
+    #    limitations under the License.
+    ################################################################################
+
+===========
+HPCC Source
+===========
+
+The most upto date details of building the system are found on the HPCC Wiki at
```
Comment (richardkchapman) [inline devdoc/Development.rst]: up-to-date

File: devdoc/Development.rst [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    sudo apt-get install fop
+
+**NOTE:** Installing the above via alternative methods (i.e. from source) may place installations outside of searched
+paths.
+
+Building the system
+===================
+
+The HPCC system is built using the cross-platform build tool cmake, which is available for Windows, virtually all
+flavors of Linux, FreeBSD, and other platforms. You should install cmake version 2.8.3 or later before building the
```
Comment (richardkchapman) [inline devdoc/Development.rst]: I think we need later than 2.8.3 these days

**Comment by richardkchapman** [comment]
@ghalliday A couple of trivial comments but I'll happily merge anyway

**Comment by JamesDeFabia** [comment]
@ghalliday I agree with @richardkchapman . This should be merged and we (you) can address the trivial changes later

**Pull request merged**